### PR TITLE
[MIRROR] [Ready] MODsuits [MDB IGNORE]

### PR DESCRIPTION
**Pull request by SkyratBot**
Original PR: https://github.com/tgstation/tgstation/pull/59109
--------------------

## About The Pull Request

Adds Modular Suits, a 1+ year old project of replacing hardsuits with suits you can completely customize with different modules, read https://hackmd.io/@ tgstation/SypLCs4jU for more information.
### I'm here from testmerge! How do stuff!
Print parts from exosuit fabricators, buy a pack of cores from cargo in the science category. (alternatively you can scour maint, craft it with an ethereal heart or pray to gods)
Examine the shell for instructions, build the suit.
Screwdriver to open the suit, put power cell in, put any modules in, crowbar to take modules out.
Put on back, extend parts with action buttons and activate.
Good luck!

cool guy here:
image
showcasing the ui with the dynamic configuration system:
image

Closes #63056
Closes #62607
Closes #55846
Closes #50265
Closes #48402
Closes #44536 
Closes #40874
Closes #36986
Closes #27966

## Why It's Good For The Game

It allows more customization, adds a more robust system for special hardsuit stuff, lets us do much more creative things and encourages the players to be more creative 

## Changelog
:cl: Fikou; Sprites: PositiveEntropy, Azlan, Infrared_Baron, Thunder12345, Yeeye; Code: Waltermeldron, Goof, Armhulen, ArcaneMusic, DatBoiTim; Descriptions: Nerevar
add: Adds Modular Suits. Printable at robotics.
del: Removes Hardsuits.
balance: Almost all Hardsuits have been replaced with MODsuits.
balance: HECKsuit, Berserker Hardsuit, Cult Hardsuit and Carp Hardsuit became hoodies, the first 2 losing their spaceproofness
del: Cybersun, Combat Medic and Owl Hardsuits didn't make the cut.
refactor: Changes how mob gravity checks are made. Uses a gravity negation trait. As part of this, space wind pushing you is handled by gravity negation, not noslips.
balance: Hardsuits no longer defend from tackles, but negating gravity does.
code: Adds support for digitigrade compatible shoes.
qol: Piercing syringes now pierce when used from hand, not just as projectile.
code: Removes the unused outside of admin shenanigans cult hardsuit requirement for spells.
/:cl:

### CODE REVIEW

**Comment by press-mark** [comment]
real

**Comment by Ebin-Halcyon** [comment]
Aw fuck boys and girls.

**Comment by GoldenAlpharex** [comment]
*Just a week away! Can you believe it guys? MODsuits 2021 by Fikou is in a week!*

**Comment by axietheaxolotl** [comment]
AW HELL YEA

**Comment by Stalkeros2** [comment]
LET'S GOOOOOOO

**Comment by Fikou** [comment]
glhf

File: modular_skyrat/modules/altborgs/code/robot_model.dm [hunk 1]
```diff
@@ -374,7 +374,7 @@
 	model_select_icon = "ninjaborg"
 	model_select_alternate_icon = 'modular_skyrat/modules/altborgs/icons/screen_cyborg.dmi'
 	model_traits = list(TRAIT_PUSHIMMUNE, TRAIT_NOFLASH)
-	magpulsing = TRUE
+	model_traits = list(TRAIT_NEGATES_GRAVITY)
```
Comment (Fikou) [inline modular_skyrat/modules/altborgs/code/robot_model.dm:377]: yeah

File: modular_skyrat/modules/cargo/code/packs.dm [hunk 2]
```diff
@@ -232,20 +232,20 @@
 	cost = CARGO_CRATE_VALUE * 18
 	crate_name = "advanced extinguisher crate"
 
-/datum/supply_pack/engineering/hardsuit_engineering
-	name = "Engineering Hardsuit Crate"
+/datum/supply_pack/engineering/modsuit_engineering
+	name = "Engineering MODSUIT Crate"
```
Comment (Fikou) [inline modular_skyrat/modules/cargo/code/packs.dm]: MODsuit

File: modular_skyrat/modules/cargo/code/packs.dm [hunk 3]
```diff
@@ -232,20 +232,20 @@
...
-	contains = list(/obj/item/clothing/suit/space/hardsuit/engine)
+	contains = list(/obj/item/mod/control/pre_equipped/engineering)
 	cost = CARGO_CRATE_VALUE * 13
 	crate_name = "engineering hardsuit crate"
 	crate_type = /obj/structure/closet/crate/secure/engineering
 
-/datum/supply_pack/engineering/hardsuit_atmospherics
-	name = "Atmospherics Hardsuit Crate"
+/datum/supply_pack/engineering/modsuit_atmospherics
+	name = "Atmospherics MODSUIT Crate"
```
Comment (Fikou) [inline modular_skyrat/modules/cargo/code/packs.dm]: MODsuit

**Pull request merged**